addons/survey/models/survey_survey.py

```diff
@@ -204,13 +204,18 @@ def _create_answer(self, user=False, partner=False, email=False, test_entry=Fals
             else:
                 answer_vals['email'] = email
 
+            if attempts_pool_token is True:
+                answer_vals['attempts_pool_token'] = str(uuid.uuid4())
+            elif attempts_pool_token:
```

Comment:
Here is what I come up with:

`if force_invite_token:
    answer_vals['invite_token'] = force_invite_token
elif survey.is_attempts_limited and survey.access_mode != 'public':
    # attempts limited: create a new invite_token
    # exception made for 'public' access_mode since the attempts pool is global because answers are
    # created every time the user lands on '/start'
    answer_vals['invite_token'] = str(uuid.uuid4())`

How does it look?

Reply:
Anyway, should be good now as discussed with TDE (hopefully).